deps/chakrashim/src/jsrtstringutils.cc

```diff
@@ -23,175 +23,175 @@
-        break;
-      }
-    }
-  }
-
-  // if nothing could be written, just leave
-  if (strLengthToWrite == 0) {
-    if (bytesWritten != nullptr) {
-      *bytesWritten = 0;
-    }
-
-    if (charsWritten != nullptr) {
-      *charsWritten = 0;
-    }
-
-    return JsNoError;
-  }
-
-  // Do the actual conversion
-  DWORD result = WideCharToMultiByte(code,
-                                     0,
-                                     str,
-                                     static_cast<int>(strLengthToWrite),
-                                     buffer,
-                                     static_cast<int>(bufferSize),
-                                     NULL,
-                                     NULL);
-
-  if (result == 0) {
-    DWORD conversionError = GetLastError();
-    if (conversionError != ERROR_INSUFFICIENT_BUFFER) {
-      // This should never happen.
-      return JsErrorFatal;
-    }
-  }
-
-  if (bytesWritten != nullptr) {
-    *bytesWritten = result;
-  }
-
-  if (charsWritten != nullptr) {
-    *charsWritten = strLengthToWrite;
-  }
-
-  return JsNoError;
-}
-
-JsErrorCode StringConvert::InternalToWChar(const char *str,
-                                           const size_t length,
-                                           const int code,
-                                           wchar_t* buffer,
-                                           size_t size,
-                                           __out size_t *charsWritten) {
-  //
-  // Special case of empty input string
-  //
-  if (str == nullptr || length == 0) {
-    return JsNoError;
-  }
-
-  //
-  // Get length (in wchar_t's) of resulting UTF-16 string
-  //
-  const int utf16Length = ::MultiByteToWideChar(
-    code,                      // convert the given code
-    0,                         // flags
-    str,                       // source string
-    static_cast<int>(length),  // length (in chars) of source string
-    NULL,                      // unused - no conversion done in this step
-    0);           // request size of destination buffer, in wchar_t's
-
-  if (utf16Length == 0) {
-    // Error
-    return JsErrorFatal;
-  }
-
-  if (size == -1) {
-    size = utf16Length;
-  }
-
-  //
-  // Do the conversion from code to UTF-16
-  //
-  *charsWritten = ::MultiByteToWideChar(
-    code,                       // convert from code
-    0,                          // validation was done in previous call,
-                                // so speed up things with default flags
-    str,                        // source string
-    static_cast<int>(length),   // length (in chars) of source string
-    buffer,                     // destination buffer
-    static_cast<int>(size));    // size of destination buffer, in wchar_t's
-
-  if (*charsWritten == 0) {
-    // Error
-    return JsErrorFatal;
-  }
-
-  //
-  // Return resulting UTF-16 string
-  //
-  return JsNoError;
-}
+// JsErrorCode StringConvert::GetCharLength(const wchar_t *str,
```

Comment:
Yes, I saw your commit message, but i was saying it would make more sense to add that comment in the file as well or have `#ifdef` for win32.